### PR TITLE
Implement recent searches

### PR DESCRIPTION
**Pull request by thagikura**
- Add Repository and Dao for recent searches
- Add recent searches body in the Search UI
- Add relevant tests

Screenshots:

### CODE REVIEW

**Comment by thagikura** [comment]
The same change as #670, but creating a new PR as I had to resolve  a lot of conflicts in #670.

File: feature/search/src/main/java/com/google/samples/apps/nowinandroid/feature/search/SearchViewModel.kt [hunk 1]
```diff
@@ -70,9 +75,37 @@ class SearchViewModel @Inject constructor(
             initialValue = SearchResultUiState.Loading,
         )
 
+    val recentSearchQueriesUiState: StateFlow<RecentSearchQueriesUiState> =
+        recentSearchQueriesUseCase().map {
```
Comment (mmoczkowski) [inline feature/search/src/main/java/com/google/samples/apps/nowinandroid/feature/search/SearchViewModel.kt:79]: NIT: 
`map(RecentSearchQueriesUiState::Success)`

**Review by mmoczkowski — APPROVED** [review]
Great feature, just a couple of tiny nits, feel free to ignore

File: core/testing/src/main/java/com/google/samples/apps/nowinandroid/core/testing/repository/TestRecentSearchRepository.kt [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+import kotlinx.coroutines.flow.Flow
+import kotlinx.coroutines.flow.flow
+
+class TestRecentSearchRepository : RecentSearchRepository {
+
+    private val cachedRecentSearches: MutableList<RecentSearchQuery> = mutableListOf()
+
+    override fun getRecentSearchQueries(limit: Int): Flow<List<RecentSearchQuery>> {
+        return flow {
+            emit(cachedRecentSearches.sortedByDescending { it.queriedDate }.take(limit))
```
[the remark this answers is not in the document]
Comment (thagikura) [reply core/testing/src/main/java/com/google/samples/apps/nowinandroid/core/testing/repository/TestRecentSearchRepository.kt]: 1: Done.
2: It's ok to use flowOf. I think it produces the same result. Maybe flowOf is a bit simpler. Changed this to flowOf

File: feature/search/src/androidTest/java/com/google/samples/apps/nowinandroid/feature/search/SearchScreenTest.kt [hunk 3]
```diff
@@ -153,4 +153,29 @@ class SearchScreenTest {
...
     }
+
+    @Test
+    fun emptyQuery_notEmptyRecentSearches_verifyClearSearchesButton_displayed() {
+        val recentSearches = listOf("kotlin", "testing")
+        composeTestRule.setContent {
+            SearchScreen(
+                searchResultUiState = SearchResultUiState.EmptyQuery,
+                recentSearchesUiState = RecentSearchQueriesUiState.Success(
+                    recentQueries = recentSearches.map {
```
[the remark this answers is not in the document]
Comment (thagikura) [reply feature/search/src/androidTest/java/com/google/samples/apps/nowinandroid/feature/search/SearchScreenTest.kt]: Done.

File: feature/search/src/main/java/com/google/samples/apps/nowinandroid/feature/search/SearchViewModel.kt (hunk 1 above)
Comment (thagikura) [reply feature/search/src/main/java/com/google/samples/apps/nowinandroid/feature/search/SearchViewModel.kt]: Done.

**Pull request merged**